src/frame_buffer.rs

```diff
@@ -84,11 +103,12 @@ impl<'a> FrameBuffer<'a> {
             );
         }
         let width = self.dimensions.0;
+        let origin_offset = self.origin_offset::<T>();
         unsafe {
             self.insert_raw(
                 name,
                 T::pixel_type(),
-                data.as_ptr() as *const c_char,
+                (data.as_ptr() as *const c_char).offset(origin_offset),
```

Comment:
Can you explain the logic behind this offset, and the similar one below?

Reply:
From the behaviour observed I believe that the underlying API expects the data passed here, in insert_raw to be pointing at the pixel at (0,0) rather than the "first" pixel in the framebuffer.

As the reason why, no idea. I've seen this pattern in other APIs though.

Reply:
> the pixel mapped to (0,0) in the data_window rather than the "first" pixel in the framebuffer.

I'm not sure what this means. Are you talking about the pixel at coordinates (0, 0) relative to the origin of the data window, or of the display window? What is the "'first'" pixel?

Reply:
@ralith sounds very sensible - just let me know if you need further changes in the branch.